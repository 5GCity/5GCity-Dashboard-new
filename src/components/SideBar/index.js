/**
 * SideBar Component
 * Please write a description
 *
 * @author Guilherme Patriarca <gpatriarca@ubiwhere.com>
 */
import styled from 'styled-components'


const SideBar = styled.nav`
  position: absolute;
  margin:0;
  width: 120px;
  display: block;
  background-color: ${({ theme }) => theme.bodyBackground};
  box-shadow: -5px 0 20px 0 rgba(0,0,0,0.5);
  height: 100%;
`
const Container = styled.ul`
  color: white;
  text-align: center;
  cursor: pointer;
  margin: 0;
  padding:0;
  list-style: none;
`
const Menu = styled.li`
  text-transform: capitalize;
  display: flex;
  justify-content:center;
  align-content:center;
  flex-direction:column;
  padding: 14px 16px;
  height: ${({height}) => height}px;
  box-shadow: inset 0 -1px 0 0 rgba(137,151,159,0.15);
  &:hover {
    border-right: 3px solid ${({ theme }) => theme.primaryColor};
  }
  ${({active, theme }) => active && `
    border-right: 3px solid ${theme.primaryColor};
  `}
  ${({disabled}) => disabled && `
    cursor: not-allowed;
    color: rgba(255,255,255,0.3);
  `}
`
const SubMenu = styled.ul`
  text-transform: capitalize;
  box-shadow: inset 0 -1px 0 0 rgba(137,151,159,0.15);
  cursor: pointer;
  margin: 0;
  padding:0;
  list-style: none;
  display: flex;
  justify-content:center;
  align-content:center;
  flex-direction:column;
  width: 100%;
  background-color: rgba(0,0,0,0.1);
  ${({show}) => !show &&`
    visibility: hidden;
    opacity: 0;
    left: 0;
    transition: opacity 0.5s ease;
    height: 0px;
  `}
  ${({show}) => show &&`
  visibility: visible;
  opacity: 1;
  transition: opacity 0.5s ease;
  height: 40px;
`}
  &:hover {
    border-right: 3px solid ${({ theme }) => theme.primaryColor};
  }
  ${({disabled}) => disabled && `
  cursor: not-allowed;
  color: rgba(255,255,255,0.3);
`}
${({active, theme }) => active && `
  border-right: 3px solid ${theme.primaryColor};
`}
`

const Link = styled.a`
  display:block;
  font-size: 14px;
  font-weight: bold;
  font-family:${({ theme }) => theme.fontDin};
  position: relative;
`

const Icon = styled.i`
  font-size: 32px;
`

const UserInfo = styled.div`
  position: absolute;
  bottom: 0;
  width: 100%;
  padding: 18px 8px;
  background-color: rgba(255,255,255,0.02);
  box-shadow: inset 0 1px 0 0 rgba(137,151,159,0.15);
`
const UserName = styled.p`
  font-family: "Open Sans";
  margin: 0 0 6px 0;
  color: #FFFFFF;
  font-size: 12px;
  line-height: 12px;
`
const UserRole = styled.p`
  font-family: "Open Sans";
  margin: 0;
  font-size: 12px;
  line-height: 12px;
  color: #8CC14E;
  font-weight: bold;
`
const IconUser = styled.i`
  position: absolute;
  right: 0;
  top: 40%;
  cursor: pointer;
  transform: rotate(90deg);
  color: #fff;
`

SideBar.Container = Container
SideBar.Menu = Menu
SideBar.SubMenu = SubMenu
SideBar.Link = Link
SideBar.Icon = Icon
SideBar.UserInfo = UserInfo
SideBar.UserName = UserName
SideBar.UserRole = UserRole
SideBar.IconUser = IconUser

export default SideBar
