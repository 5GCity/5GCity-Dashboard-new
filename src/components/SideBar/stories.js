/**
 * SideBar Component Stories
 * Please write a description or remove this line
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React, { Component } from 'react'
import { storiesOf } from '@storybook/react'
import styled from 'styled-components'

import { Icon } from 'element-react'
import { MapIcon } from 'components/Icons'
import Brand from 'components/Brand'

import SideBar from './index'

const LINKS = [
  { id: 1, path: '/infoManagement', name: 'Info Management', icon: null, disabled: false , show: ['Inf. Owner'] ,active: false},
  { id: 2, path: '/slices', name: 'Slices', icon:/*<MapIcon />*/ null, disabled: false , show: ['Inf. Owner','Slice Requester'] ,active: true},
  { id: 3, path: '/network', name: 'Network Services', icon: null, disabled:true , show: ['Inf. Owner','Slice Requester'] ,active: false},
  { id: 4, path: '/catalogue', name: 'Catalogue', icon: null, disabled:false , show: ['Inf. Owner','Slice Requester'] ,active: false},
  { id: 5, path: '/sdk/services', name: 'SDK', icon: null, disabled:false , show: ['Slice Requester'] ,children: [
    {
    path: '/sdk/services',
    name: 'service'
  },
  {
    path: '/sdk/functions',
    name: 'function'
  }
] ,active: false
},
{ id: 6, path: '/sdk/other', name: 'My Repository', icon: null, disabled:false , show: ['Slice Requester'] ,children: [
  {
  path: '/sdk/wazaaa',
  name: 'services',
},
{
  path: '/sdk/wazaaa',
  name: 'functions',
}
] ,active: false
},
]


class SideBarComponent extends Component {

  state = {
    data: LINKS
  }


  toogleList = item => {
    const dataCopy = this.state.data
    dataCopy.map(obj => {
      if(obj.id === item.id){
        obj.active = !obj.active
      } else if (item.disabled) {
        obj.active = false
      } else {
        obj.active = false
      }
    })
    this.setState({data: dataCopy})
  }
  render () {
    return(
      <SideBar>
      <Brand />
      {this.state.data && this.state.data.map(menu =>
      <SideBar.Container key={menu.id}>
        <SideBar.Menu
          height={60}
          onClick={() => this.toogleList(menu) }
          active={menu.active}
          disabled={menu.disabled}
        >
          {menu.icon &&
          <SideBar.Icon>{menu.icon}</SideBar.Icon>
          }
          <SideBar.Link>{menu.name}</SideBar.Link>
          {menu.children && !menu.active &&
          <Caret name={'caret-bottom'} />
          }
          {menu.children && menu.active &&
          <Caret name={'caret-top'} />
          }
        </SideBar.Menu>
        {menu.children && menu.children.map((submenu, i) =>
        <SideBar.SubMenu
          key={i}
          show={menu.active}
        >
            <SideBar.Link>{submenu.name}</SideBar.Link>
        </SideBar.SubMenu>
        )}
      </SideBar.Container>
      )}
      <SideBar.UserInfo>
      <SideBar.UserName>Name Name</SideBar.UserName>
            <SideBar.UserRole>Slice Requester</SideBar.UserRole>
            <SideBar.IconUser
              className="el-icon-more">
            </SideBar.IconUser>
      </SideBar.UserInfo>
    </SideBar>
    )
  }
}

const componentSidebar = () => (
  <SideBarComponent />
  )

storiesOf('SideBar', module)
  .add('default', componentSidebar)


const Caret = styled(Icon)`
  padding-top: 2px;
  color:white;
  font-size: 12px;
`
