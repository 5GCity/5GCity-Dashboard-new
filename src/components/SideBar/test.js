/**
 * SideBar Component Tests
 *
 * @author Your Name <youremail@ubiwhere.com>
 */
import React from 'react'
import renderer from 'react-test-renderer'
import SideBar from './index'
import 'jest-styled-components'

it('renders correctly', () => {
  const tree = renderer
    .create(<SideBar />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
