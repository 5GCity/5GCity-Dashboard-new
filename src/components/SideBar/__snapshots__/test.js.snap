// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  position: absolute;
  margin: 0;
  width: 120px;
  display: block;
  box-shadow: -5px 0 20px 0 rgba(0,0,0,0.5);
  height: 100%;
}

<nav
  className="c0"
/>
`;
